plugin/session_buffer.py

```diff
@@ -854,7 +854,10 @@ def _on_code_lenses_async(self, response: list[CodeLens] | None) -> None:
                 else:
                     self.session.check_log_unsupported_command(command_name)
         for sv in self.session_views:
-            sv.handle_code_lenses_async(supported_code_lenses)
+            if supported_code_lenses:
+                sv.handle_code_lenses_async(supported_code_lenses)
+            else:
+                sv.clear_code_lenses_async()
```

Comment:
Or this `clear_code_lenses_async()` call could instead be done from `handle_code_lenses_async()`. I feel that it would make the logic more centralized and easier to follow.